### ISSUE TITLE
Release notes - swiftlint version pinning rule

### ISSUE THREAD

**Issue by Panajev** [realm/SwiftLint#2502]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The actual PR talks about an opt-in rule that would make the lint command fail if the version has not been pinned. The release notes make it seem like it is not optional but mandatory (which seems odd as it would be a breaking change for many):
```
The lint command now exits with a code of 2 when not using pinned
version defined as swiftlint_version in the configuration file.
Kim de Vos
#2074
```

**Comment by kimdv** [comment]
I agree a bit here. 

My first idea was to use a min version.

**Comment by jpsim** [comment]
> The actual PR talks about an opt-in rule that would make the lint command fail if the version has not been pinned. The release notes make it seem like it is not optional but mandatory (which seems odd as it would be a breaking change for many):

I think you're talking about the `swiftlint_version` configuration key that was introduced in #1134, first released in [version 0.16.0 in January 2017](https://github.com/realm/SwiftLint/releases/tag/0.16.0).

The changelog entry you mentioned does change the behavior of those using this feature (i.e. specifying a `swiftlint_version` entry in your configuration file) by exiting the linter immediately instead of simply logging and continuing as normal. Since in some configurations, SwiftLint can lint _a lot_, I suspect these warning logs were hard to catch.

I don't think this is a commonly used feature and therefore don't think this is a breaking change for many.

Please let me know if I've misunderstood you.
